src/watcher.js

```diff
@@ -115,41 +119,174 @@ async function getLastBlockToProcess() {
   return lastBlockNumber.sub(requiredBlockConfirmations)
 }
 
+async function processOne(
+  sendToQueue,
+  bridgeContractAddress,
+  eventContractAddress,
+  homeBridgeAddress,
+  foreignBridgeAddress,
+  eventFilter,
+  lastProcessedBlock
+) {
+  logger.debug(
+    `processOne --> bridgeContractAddress: ${bridgeContractAddress}, eventContractAddress: ${eventContractAddress}, homeBridgeAddress: ${homeBridgeAddress}, foreignBridgeAddress: ${foreignBridgeAddress}, eventFilter: ${eventFilter}, lastProcessedBlock: ${lastProcessedBlock}`
+  )
+
+  const bridgeContract = new web3Instance.eth.Contract(config.bridgeAbi, bridgeContractAddress)
+  const eventContract = new web3Instance.eth.Contract(config.eventAbi, eventContractAddress)
+
+  const lastBlockToProcess = await getLastBlockToProcess(bridgeContract)
+
+  if (lastBlockToProcess.lte(lastProcessedBlock)) {
+    logger.debug('All blocks already processed')
+    return lastBlockToProcess
+  }
+
+  const fromBlock = lastProcessedBlock.add(ONE)
+  const toBlock = lastBlockToProcess
+
+  const events = await getEvents({
+    contract: eventContract,
+    event: config.event,
+    fromBlock,
+    toBlock,
+    filter: eventFilter
+  })
+  logger.info(
+    `Found ${events.length} ${config.event} events for contract address
+      ${eventContract.options.address}`
+  )
+
+  if (events.length) {
+    const job = await processEvents(events, homeBridgeAddress, foreignBridgeAddress)
+    logger.info('Transactions to send:', job.length)
+
+    if (job.length) {
+      await sendToQueue(job)
+    }
+  }
+
+  return lastBlockToProcess
+}
+
+async function getDeployedBridges() {
+  return redis
+    .smembers(config.deployedBridgesRedisKey)
+    .then(deployedBridges => deployedBridges.map(bridge => JSON.parse(bridge)))
+}
+
 async function main({ sendToQueue }) {
   try {
-    const lastBlockToProcess = await getLastBlockToProcess()
+    if (config.id.startsWith('erc-erc-multiple')) {
```

Comment:
Can we extract the logic inside this `if` into a separate function? In that way the `main` method will be easier to read